src/dfx/src/commands/start.rs

```diff
@@ -321,7 +327,11 @@ pub fn exec(
         replica_config
     };
 
-    let effective_config = CachedConfig::replica(&replica_config);
+    let effective_config = if pocketic {
+        CachedConfig::pocketic()
+    } else {
+        CachedConfig::replica(&replica_config)
+    };
 
     if !clean && !force && previous_config_path.exists() {
```

Comment:
Since the PocketIC doesn't persist any data, this will need to reset the network id, so that any projects will reset their local canister id store.

To see the problem, try this procedure:
1. `dfx start --pocketic --background --clean`
2. `dfx deploy` (notice this succeeds)
3. `dfx stop`
4. `dfx start --pocketic --background`
5. `dfx deploy`

notice the second deploy fails:
```
Caused by: The replica returned a rejection error: reject code DestinationInvalid, reject message Canister txyno-ch777-77776-aaaaq-cai not found, error code Some("IC0301")
```
Expected behavior: `dfx deploy` creates new canisters and deploys successfully